server/src/test/java/org/opensearch/gateway/PrimaryShardBatchAllocatorTests.java

```diff
@@ -322,6 +323,21 @@ public void testAllocateUnassignedBatchOnTimeoutWithNoShards() {
         assertEquals(0, ignoredShards.size());
     }
 
+    public void testAllocateUnassignedBatchOnTimeoutSkipIgnoringNewPrimaryShards() {
```

Comment:
Should we add a test on ignored being non zero?

Reply:
This test is already present - `testAllocateUnassignedBatchOnTimeoutWithMatchingPrimaryShards`